src/io/aviso/taxi_toolkit/ui.clj

```diff
@@ -51,24 +55,30 @@
 (defn a-click
   "Element-agnostic. Runs either (taxi/click) or (click-anything)."
   [& el-spec]
-  (let [el (apply $ el-spec)]
-    (case (.getTagName (:webelement el))
-      ("a" "button") (retry #(click el))
-      (retry #(click-non-clickable el)))))
+  (retrying
+   (let [el (apply $ el-spec)]
+     (case (.getTagName (:webelement el))
+       ("a" "button") (click el)
+       (click-non-clickable el)))))
 
 (defn a-text
   "For non-form elements such as <div> works like (taxi/text).
   For <input> works like (taxi/value)."
   [el]
-  (case (.getTagName (:webelement el))
-    ("input") (retry #(value el))
-    (retry #(text el))))
+  ;; TODO: this one is sometimes used as a checker in smart-wait/wait-until
+  ;; and sometimes as a standalone operation. The retrying here is not needed
+  ;; in case of smart-wait
+  (retrying
```

Comment:
Yeah, I think this particular helper kind of doesn't fit the API: we've got finders,waiters and actions that work on `el-spec` and assertions that work on `el`.

I think I've unnecesarily altered it, the utils/retry helper is in place still, so I could just left it as it is, what do you think?